pkg/traceql/engine.go

```diff
@@ -232,3 +343,34 @@ func (s Static) asAnyValue() *common_v1.AnyValue {
 		},
 	}
 }
+
+// Regex to extract matchers from a query string
+// This regular expression matches a string that contains three groups separated by operators.
+// The first group is a string of alphabetical characters, dots, and underscores.
+// The second group is a comparison operator, which can be one of several possibilities, including =, >, <, and !=.
+// The third group is one of several possible values: a string enclosed in double quotes,
+// a number with an optional time unit (such as "ns", "ms", "s", "m", or "h"),
+// a plain number, or the boolean values "true" or "false".
+// Example: "http.status_code = 200" from the query "{ .http.status_code = 200 && .http.method = }"
+var re = regexp.MustCompile(`[a-zA-Z._]+\s*[=|<=|<=|=~|!=|>|<|!~]\s*(?:"[a-zA-Z._]+"|[0-9smh]+|true|false)`)
+
+// extractMatchers extracts matchers from a query string and returns a string that can be parsed by the storage layer.
+func extractMatchers(query string) string {
```

Comment:
it seems unlikely this is solvable with a regex. what about:

```
{ (<cond> || <cond>) && <cond> } // we can nest ()'s infinitely with booleans operators in all kinds of combinations
```

or 

```
{ <some conditions> } && { <different conditions> } 
```

I think some of these edge cases will always remain unsolved which is fine, but picking up all matchers out of a query string with a regex will provide very wrong results in a lot of cases.

Reply:
We're only targeting support for queries of the type `{ a = b && c = d && .. }`, which matches what the UI editor supports.

Every other query will probably remain unsupported until a parser in the frontend (Grafana) exists. Or a better alternative in Tempo is found.

Reply:
Those cases could be handled if OR `||` or more that one selector is found.

Reply:
Is there somewhere that we assert that the query matches this shape and if it doesn't fall back to the old behavior?
```
{ a = b && c = d && .. }
```

Reply:
Not yet. But I was planning in adding it if we're ok with the approach.

Reply:
Done in 3ca8ba6ee